packages/teams-js/src/private/filesExtensionsFor3PStorageproviders.ts

```diff
@@ -0,0 +1,194 @@
+  /**
+   * Object used to represent a file
+   * @beta
+   *
+   */
+  export interface FilesFor3PApps extends Blob {
+    /**
+     * A number that represents the number of milliseconds since the Unix epoch
+     */
+    lastModified: number;
+    /**
+     * Name of the file
+     */
+    name: string;
+    /**
+     * file type
+     */
+    type: string;
+    /**
+     * A string containing the path of the file relative to the ancestor directory the user selected
+     */
+    webkitRelativePath?: string;
+  }
+  export interface FileChunk {
+    /**
+     * Base 64 data for the requested uri
+     */
+    chunk: string;
+    /**
+     * chunk sequence number
+     */
+    chunkSequence: number;
+  }
+  export interface FileResult {
+    /**
+     * error encountered in getDragAndDropFiles API
+     */
+    error?: SdkError;
+    /**
+     * File chunk which will be assemebled and converted into a blob
+     */
+    fileChunk: FileChunk;
+    /**
+     * File index of the file for which chunk data is getting recieved
+     */
+    fileIndex: number;
+    /**
+     * File type which is getting recieved
+     */
+    fileType: string;
+    /**
+     * Will tell if this is the last file
+     */
+    isLastFile: boolean;
+    /**
+     * Will tell the name of the file
+     */
+    fileName: string;
+  }
+  export interface AttachmentListHelper {
+    fileType: string;
+    assembleAttachment: AssembleAttachment[];
+  }
+  export interface AssembleAttachment {
+    /** A number representing the sequence of the attachment in the file chunks. */
+    sequence: number;
+    /** A Blob object representing the data of the file chunks. */
+    file: Blob;
+  }
+
+  /**
+   * Get drag-and-drop files using a callback.
+   *
+   * @param dragAndDropInput - Input for drag-and-drop action.
+   * @param callback - Callback function to handle the result.
+   */
+  export function getDragAndDropFilesHandler(
+    dragAndDropInput: string,
+    callback: (attachments: FilesFor3PApps[], error?: SdkError) => void,
+  ): void {
+    if (!callback) {
+      throw new Error('[getDragAndDropFiles] Callback cannot be null');
+    }
+
+    ensureInitialized(runtime, FrameContexts.content, FrameContexts.task);
+
+    if (!dragAndDropInput || dragAndDropInput === '') {
+      //condition changed
+      const invalidInput: SdkError = { errorCode: ErrorCode.INVALID_ARGUMENTS };
+      callback([], invalidInput);
+      return;
+    }
+    getFilesDragAndDropViaCallback(dragAndDropInput, callback);
+  }
+
+  function getFilesDragAndDropViaCallback(
+    dragAndDropInput: string,
+    callback: (attachments: FilesFor3PApps[], error?: SdkError) => void,
+  ): void {
+    const files: FilesFor3PApps[] = [];
+    let helper: filesExtensionsFor3PStorageproviders.AttachmentListHelper = {
+      fileType: '',
+      assembleAttachment: [],
+    };
+
+    function handleGetDragAndDropFilesCallbackRequest(
+      fileResult: filesExtensionsFor3PStorageproviders.FileResult,
```

Comment:
No need of including namespace here as the `FileResult` is within this namespace.